kbchat/kbchat.go

```diff
@@ -580,6 +580,77 @@ func (a *API) GetUsername() string {
 	return a.username
 }
 
+func (a *API) ListChannels(teamName string) ([]string, error) {
+	apiInput := fmt.Sprintf(`{"method": "listconvsonname", "params": {"options": {"topic_type": "CHAT", "members_type": "team", "name": "%s"}}}`, teamName)
+	output, err := a.doFetch(apiInput)
+	if err != nil {
+		return nil, err
+	}
+
+	var channelsList ChannelsList
+	inboxRaw := output.Text()
+	if err := json.Unmarshal([]byte(inboxRaw[:]), &channelsList); err != nil {
+		return nil, err
+	}
+
+	var channels []string
+	for _, conv := range channelsList.Result.Convs {
+		channels = append(channels, conv.Channel.TopicName)
+	}
+	return channels, nil
+}
+
+func (a *API) JoinChannel(teamName string, channelName string) error {
+	apiInput := fmt.Sprintf(`{"method": "join", "params": {"options": {"channel": {"name": "%s", "members_type": "team", "topic_name": "%s"}}}}`, teamName, channelName)
+	_, err := a.doFetch(apiInput)
+	return err
+}
+
+func (a *API) ListMembersOfTeam(teamName string) (ListTeamMembersResultMembers, error) {
+	empty := ListTeamMembersResultMembers{}
+
+	apiInput := fmt.Sprintf(`{"method": "list-team-memberships", "params": {"options": {"team": "%s"}}}`, teamName)
+	cmd := a.runOpts.Command("team", "api")
```

Comment:
can we pull the ones that are associated with team api out of kbchat.go and into a new file called `team.go`? maybe also include the types in that file. like wallet.go.

Reply:
Done